From 9e964794da43ca5a448039986d93901ba2a0695e Mon Sep 17 00:00:00 2001
From: Jiang Lu <lu.jiang@windriver.com>
Date: Thu, 15 Sep 2011 15:49:44 +0800
Subject: [PATCH] 85xx:Powerpc:L2SRAM:Fix improper address translation routine

Since the 85xx cpu support 36bit physical address, the data type of
physical address should be phys_addr_t. Therefore, we should use
strict_strtoull() to translate string into physical address.

Signed-off-by: Jiang Lu <lu.jiang@windriver.com>
Signed-off-by: Xufeng Zhang <xufeng.zhang@windriver.com>
---
 arch/powerpc/sysdev/fsl_85xx_l2ctlr.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/powerpc/sysdev/fsl_85xx_l2ctlr.c b/arch/powerpc/sysdev/fsl_85xx_l2ctlr.c
index 15cc380..b58e089 100644
--- a/arch/powerpc/sysdev/fsl_85xx_l2ctlr.c
+++ b/arch/powerpc/sysdev/fsl_85xx_l2ctlr.c
@@ -41,11 +41,11 @@ static long get_cache_sram_size(void)
 	return val;
 }
 
-static long get_cache_sram_offset(void)
+static phys_addr_t get_cache_sram_offset(void)
 {
-	unsigned long val;
+	unsigned long long val;
 
-	if (!sram_offset || (strict_strtoul(sram_offset, 0, &val) < 0))
+	if (!sram_offset || (strict_strtoull(sram_offset, 0, &val) < 0))
 		return -EINVAL;
 
 	return val;
-- 
1.7.0.4

